client/state/data-layer/wpcom/videos/poster/index.js

```diff
@@ -0,0 +1,51 @@
+/**
+ * Internal dependencies
+ */
+import { http } from 'state/data-layer/wpcom-http/actions';
+import { dispatchRequest } from 'state/data-layer/wpcom-http/utils';
+import { VIDEO_EDITOR_POSTER_UPDATE } from 'state/action-types';
+import {
+	updateVideoEditorPosterSuccess,
+	updateVideoEditorPosterFailure,
+	updatingVideoEditorPoster,
+} from 'state/ui/editor/video-editor/actions';
+
+/**
+ * Updates the poster for a video.
+ *
+ * @param  {Object} store  Redux store
+ * @param  {Object} action  Action object
+ * @param {Function} next  Dispatches to next middleware in chain
+ * @returns {Object} original action
+ */
+export const updateVideoEditorPoster = ( { dispatch }, action, next ) => {
+	if ( ! ( 'file' in action.params ) && ! ( 'at_time' in action.params ) ) {
+		return next( action );
+	}
+
+	dispatch( updatingVideoEditorPoster() );
+
+	dispatch( http( {
+		apiVersion: '1.1',
+		method: 'POST',
+		path: `/videos/${ action.videoId }/poster`,
+		body: 'at_time' in action.params ? action.params : undefined,
+		formData: 'file' in action.params ? [ [ 'poster', action.params.file ] ] : undefined,
+		onSuccess: action,
+		onFailure: action,
+	} ) );
+
+	return next( action );
```

Comment:
`next()` is going to be used when we're passing the given action along the middleware chain. maybe we intercept post editor updates and want to make sure that it's `capital_P_dangit`. we could intercept that action, filter `post_content`, and then we would want to pass it along to the next link in the chain with `next()` - this is like `add_filter()` in WP.

on the other hand, we might prefer to trigger some side-effect or corresponding action. maybe every time we see `Wordpress` we want to bump some state. in this case we would pass along the original untouched action to `next()` but since we're firing off a new action altogether we want it to start at the front of the middleware stack so we'll fire off `dispatch( bumpStat( … ) )` - akin to `add_action` in WP.

@coderkevin just since we recently discussed actions and filters I thought I'd poke you on this ^^^